src/main/scala/is/hail/asm4s/AsmFunction.scala

```diff
@@ -17,3 +17,10 @@ trait AsmFunction12[T1,T2,T3,T4,T5,T6,T7,T8,T9,T10,T11,T12,R] {
 trait AsmFunction13[T1,T2,T3,T4,T5,T6,T7,T8,T9,T10,T11,T12,T13,R] {
   def apply(t1: T1, t2: T2, t3: T3, t4: T4, t5: T5, t6: T6, t7: T7, t8: T8, t9: T9, t10: T10, t11: T11, t12: T12, t13: T13): R
 }
+
+trait SpecializedOrderingFunction[T]
```

Comment:
I think you should seal this trait since it shouldn't be directly subclassed right?